lib/routing_nb_config.c

```diff
@@ -47,12 +47,14 @@ int routing_control_plane_protocols_control_plane_protocol_create(
 	case NB_EV_APPLY:
 		vrfname = yang_dnode_get_string(args->dnode, "./vrf");
 		vrf = vrf_lookup_by_name(vrfname);
-		vrf = vrf ? vrf : vrf_get(VRF_UNKNOWN, vrfname);
-		if (!vrf) {
-			flog_warn(EC_LIB_NB_CB_CONFIG_APPLY,
-				  "vrf creation %s failed", vrfname);
-			return NB_ERR;
-		}
+
+		/*
+		 * If the daemon relies on the VRF pointer stored in this
+		 * dnode, then it should register the dependency between this
+		 * module and the VRF module. If such dependency is not
+		 * registered, then this pointer might be NULL or become stale
+		 * when the VRF is deleted.
+		 */
```

Comment:
I wonder if it's still wise to check the pointer is not null here?

Reply:
Yes, it makes sense for daemons that register the dependency between the nodes. I'll think about how to implement it.

Reply:
Made the code more generic and returned the null-check.